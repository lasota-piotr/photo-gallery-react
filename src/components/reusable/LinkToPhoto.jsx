import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const LinkToPhoto = ({
  children,
  id,
  state,
}) => (
  <Link
    to={{
      pathname: `/photos/${id}`,
      state,
    }}
  >
    {children}
  </Link>
);

LinkToPhoto.propTypes = {};

export default LinkToPhoto;
