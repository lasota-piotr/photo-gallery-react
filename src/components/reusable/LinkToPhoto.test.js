import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import LinkToPhoto from './LinkToPhoto';

test("Generates anchor with url to photo's page", () => {
  const component = renderer.create(
    <MemoryRouter>
      <LinkToPhoto id="123456">Test link</LinkToPhoto>
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
