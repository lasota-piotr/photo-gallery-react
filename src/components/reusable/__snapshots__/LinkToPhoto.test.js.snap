// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates anchor with url to photo's page 1`] = `
<a
  href="/photos/123456"
  onClick={[Function]}
>
  Test link
</a>
`;
